Bug 1809642: Update dev console monitoring page use Dashboard components

jeff-phillips-18 opened the pull request:
**Fixes**: 
https://issues.redhat.com/browse/ODC-3162

**Analysis / Root cause**: 
Monitoring page was not using dashboard components

**Solution Description**: 
Use the shared Dashboard components to create the monitoring graphs

**Screenshots**:
Before:
![image](https://user-images.githubusercontent.com/11633780/75793968-1f238b00-5d3e-11ea-825e-3af4f26681f5.png)

After:
![image](https://user-images.githubusercontent.com/11633780/75794003-28145c80-5d3e-11ea-9891-8d6572e5f2b1.png)


**Browser conformance**: 
- [x] Chrome
- [x] Firefox
- [x] Safari
- [ ] Edge

/kind bug

cc @openshift/team-devconsole-ux

jeff-phillips-18 commented:
/cherrypick release-4.4

andrewballantyne commented:
/retest

andrewballantyne commented on frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboardGraph.scss:
```diff
@@ -1,6 +1,10 @@
     border: none;
-    margin: 0;
+    padding: 0;
   }
```
```
.query-browser__wrapper {
    border: 1px solid #ccc;
    margin: 0 0 20px 0;
    overflow: visible;
    padding: 10px;
    width: 100%;
}
```

Not sure we can completely kill the padding and just accept margin. The chart gets pretty close to the header. 

With no alterations (taking the padding & margin from above):
![Screen Shot 2020-03-03 at 1 20 48 PM](https://user-images.githubusercontent.com/8126518/75806611-ff965d80-5d51-11ea-957d-bf8ef257fefe.png)

And then without padding (but keeping margin):
![Screen Shot 2020-03-03 at 1 21 01 PM](https://user-images.githubusercontent.com/8126518/75806634-0cb34c80-5d52-11ea-9576-9235c803df6b.png)

Margin isn't really helping a lot, there is already whitespace under the chart, but padding looks to be quite useful... Was this just a subjective change @jeff-phillips-18?

jeff-phillips-18 replied:
Leaving the padding to 10px gives unnecessary padding around the chart. The DashboardBody component should control the spacing. Leaving the 10px padding around the chart gives this (padding in green):

![image](https://user-images.githubusercontent.com/11633780/75809347-a7ae2580-5d56-11ea-98da-60b05e0d99b6.png)

which I don't believe is desireable.

andrewballantyne replied:
Perhaps the spacing left & right is wasteful... I think the padding top has value to keep a spacer between the graph labels & the header spacer line. The bottom spacing is as useful as the margin I suspect.

Thoughts @jeff-phillips-18 ?

jeff-phillips-18 replied:
The margin used is for the bottom legend labels and I believe ought not be changed.
The spacing between the labels and the bottom of the card is part of the card styling and should be left to be consistent with other cards.

I agree there could be more top spacing. I'll look to make it consistent with other cards.

jeff-phillips-18 replied:
![image](https://user-images.githubusercontent.com/11633780/75813758-8d784580-5d5e-11ea-8804-ec7905d06fa2.png)

andrewballantyne replied:
Looking pretty good, thanks! I'll check it out in a moment to do one last pass before 👍

andrewballantyne commented:
There is also this awkward spacing between the header and the cards... likely a result of the margin shift on the buttons in the top right. We may need to adjust the styles to use `absolute` so they don't steal screen space.

![Screen Shot 2020-03-03 at 1 33 34 PM](https://user-images.githubusercontent.com/8126518/75807590-9fa0b680-5d53-11ea-90f5-dced09ee52a7.png)

andrewballantyne commented:
> There is also this awkward spacing between the tabs and the grey background... likely a result of the margin shift on the buttons in the top right. We may need to adjust the styles to use `absolute` so they don't steal screen space.
> 
> ![Screen Shot 2020-03-03 at 1 33 34 PM](https://user-images.githubusercontent.com/8126518/75807590-9fa0b680-5d53-11ea-90f5-dced09ee52a7.png)

Something like:
![image](https://user-images.githubusercontent.com/8126518/75807873-29e91a80-5d54-11ea-9ef5-39993d54d8ea.png)

```
.odc-monitoring-dashboard {
  position: relative;

  &__dropdown-options {
    align-self: flex-end;
    display: flex;
    float: right;
    position: absolute;
    right: 0;
    top: -65px;
  }
}
```
With the fragment being a div for the page.

andrewballantyne commented on frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboard.scss:
```diff
@@ -3,5 +3,7 @@
     margin-top: -65px;
+    .monitoring-dashboards__dropdown-wrap {
+      margin-bottom: 5px;
+    }
```
Personally, I am not a huge fan of using margins... this would be one reason why. Margins don't like to stay to their own element - they venture out and screw with the layout / rendering of parent elements.

I really think we should implement a `position: absolute` here as it removes it from the flow of the page -- which is what we are trying to do here.

christianvogt replied:
Whatever we do here is a bit of a hack as noted in the original PR that added these dropdowns.
The `float` is present because of how these dropdowns are used in the orignal monitoring dashboard page. They need to float to stay within the flow for responsive layout i believe.

For our purpose I agree that absolute positioning is fine.
Either way, this needs to be revisited because of the responsive layout issues we have with these dropdowns overlapping the tabs.

jeff-phillips-18 replied:
this is simply overriding the current margin such that it doesn't overflow

andrewballantyne replied:
Fair enough, let us keep it this way then... the devil we know vs the devil we don't.

jeff-phillips-18 commented:
Updated per comments, fixed for mobile resolutions:

![image](https://user-images.githubusercontent.com/11633780/75823513-e3091e00-5d6f-11ea-99e6-46489e70b9a5.png)

@openshift/team-devconsole-ux

andrewballantyne commented on frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboard.scss:
```diff
@@ -1,7 +1,29 @@
+    .monitoring-dashboards__dropdown-wrap:last-of-type {
+      margin-right: 0;
+    }
+  }
```
You can merge these two, no? Don't recall if SASS plays well... but I think it would...

andrewballantyne commented on frontend/packages/dev-console/src/components/monitoring/dashboard/MonitoringDashboard.scss:
```diff
@@ -1,7 +1,29 @@
+$no-wrap-breakpoint: 915px;
+.odc-monitoring-dashboard {
+  position: relative;
+  .odc-monitoring-dashboard__dropdown-options {
```
```suggestion
  &__dropdown-options {
```

andrewballantyne commented:
Both tests failed with this:

> 2020/03/04 00:17:53 No custom metadata found and prow metadata already exists. Not updating the metadata.
2020/03/04 00:17:55 Ran for 12s
2020/03/04 00:17:55 Submitted failure event to sentry (id=f75982005fd94500b5fd6e6503f24960)
error: could not run steps: step console failed: could not wait for build: the build console failed with reason GenericBuildFailed: Generic Build failure - check logs for details.

Hopefully this is not the sign of something bigger at work.

Going to retrigger in the hopes of a fluke.

/retest

jeff-phillips-18 commented:
/retest
